IS-828: Implement cancel transaction

boyeon555 opened the pull request:
- Adds `cancelTransaction` API
- Refactors code
- Makes failed unit tests successful
- Adds integration test for `cancelTransaction`

Spl3en committed:
Add cancelTransaction API

Spl3en committed:
Merge branch 'master' of github.com:iconation/multisig-wallet

Spl3en committed:
Add "cancelTransaction" API

Spl3en committed:
Merge branch 'master' of github.com:iconation/multisig-wallet

Spl3en committed:
Fix getTransactionCount API

boyeon555 committed:
Merge pull request #23 from iconation/master

Add "cancelTransaction" API

boyeon555 committed:
Refactor code

- Makes failed unit tests successful

boyeon555 committed:
Add integratation test for cancelTransaction

- Modifies revert msgs

Spl3en commented:
Thanks a lot @boyeon555, awesome! 👍 😊

soobokjin commented on multisig_wallet/multisig_wallet.py:
```diff
@@ -373,26 +388,28 @@ def getConfirmations(self, _offset: int, _count: int, _transactionId: int) -> li
-        for tx_id in range(_offset, _count):
-            if (_pending and not self._transactions[tx_id][0]) or (_executed and self._transactions[tx_id][0]):
+        for tx_id in range(_offset, transaction_count):
+            if len(transaction_list) == _count:
```
I guess This logic is needed to be fixed. In this logic, '_count' is used to check the count of transactions, but this is not my intention.

boyeon555 committed:
Modify code reviews

- Converts '_check_valid_transaction' to '_check_transaction_exists'
- Modifies way to check unless transaction is None

boyeon555 committed:
Modify code reviews
